src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectCapabilityResolver.cs

```diff
@@ -26,6 +27,8 @@ internal sealed class ProjectCapabilityResolver : IProjectCapabilityResolver, ID
     private readonly JoinableTaskFactory _jtf;
     private readonly CancellationTokenSource _disposeTokenSource;
 
+    private ImmutableDictionary<(string, string), bool> _cachedCapabilities = ImmutableDictionary<(string, string), bool>.Empty;
```

Comment:
Does this _really_ need to be an `ImmutableDictionary<,>`? It seems like we care more about the performance of look up and there won't be much contention, if any. Consider whether a `Dictionary<,>` would be a better choice.